ledger/simulation/simulator_test.go

```diff
@@ -48,13 +48,13 @@ func TestNonOverridenDataLedgerMethodsUseRoundParameter(t *testing.T) {
 	overridenMethods := []string{
 		"Latest",
 		"LookupLatest",
+		"LatestTotals",
```

Comment:
maybe I am pointing at a wrong position to ask the question -- it seems the methods added in ledger (acctupdates and ledger) are tied to this overriden methods, so is it because we want to implement all these because we want to implement a certain interface for ledger (I suppose ledger related with follower node) in simulator package?

Reply:
The ledger interface has two types of methods: ones that accept the round as an input param, and ones which use the latest round to look up information.

For simulate, the methods which use the latest round are problematic, for two reasons:
1. The latest round can change at any time, and it would be bad if it advanced during a simulate run (data race/inconsistent inforamtion).
2. (Now because of this PR) Simulate can run based on the state of the chain a few rounds back, so we never want to see the latest data.

I think the simulator ledger didn't override `LatestTotals` previously because it wasn't necessary at the time -- this method only gets called at the very beginning of evaluation, when it's extremely likely the round would not have advanced yet. Now we definitely need to override it, but I needed to expose a companion method, `Totals` on the underlying ledger to access the information for a given round, not just the latest.

Let me know if this helps or I missed anything.

Reply:
I see, this fills my gap in reasoning!